app/models/document.rb

```diff
@@ -1,27 +1,14 @@
-class AbstractDocument
+class Document
   attr_reader :title
 
-  def self.date_metadata_keys
-    []
-  end
-
-  def self.tag_metadata_keys
-    []
-  end
-
-  def self.metadata_keys
-    tag_metadata_keys + date_metadata_keys
-  end
-
-  def self.metadata_name_mappings
-    {}
-  end
-
-  def initialize(attrs)
+  def initialize(attrs, finder)
+    attrs = attrs.with_indifferent_access
```

Comment:
All the test data is JSON, but the hash returned in the real world is symbols so all these setters in the initialize method fail in the tests. Should I change all the data?